src/Symfony/Component/Debug/DebugClassLoader.php

```diff
@@ -522,4 +557,149 @@ private function getOwnInterfaces($class, $parent)
 
         return $ownInterfaces;
     }
+
+    private function saveReturnType(string $class, \ReflectionMethod $method, string $doc): void
+    {
+        if ($method->getReturnType() instanceof \ReflectionType) {
+            return;
+        }
+
+        if (false === strpos($doc, 'return')) {
+            return;
+        }
+
+        if (!preg_match_all('/\n\s+\* @return +(\S+)/', $doc, $matches, PREG_SET_ORDER)) {
+            return;
+        }
+
+        $types = explode('|', end($matches)[1], 3);
+
+        $nullable = false;
+
+        switch (\count($types)) {
+            case 1:
+                $type = $types[0];
+
+                break;
+            case 2:
+                // fastest way to handle the null|null edge case
+                $type = 'null';
+
+                foreach (array_unique(array_map([$this, 'normalizeType'], $types)) as $i => $normalizedType) {
+                    if ('null' === $normalizedType) {
+                        $nullable = true;
+                    } else {
+                        $type = $types[$i];
+                    }
+                }
+
+                if (!$nullable) {
+                    return;
+                }
+
+                break;
+            default:
+                return;
```

Comment:
we could have `array|iterable|anything1[]|anything2[]` trigger a deprecation telling the `iterable` type will be added